tasks/ast_tools/src/generators/typescript.rs

```diff
@@ -170,8 +170,18 @@ fn generate_ts_type_def_for_struct_field<'s>(
 ) {
     let field_type_name = if let Some(append_field_index) = field.estree.append_field_index {
         let appended_field = struct_def.fields[append_field_index].type_def(schema);
-        let appended_field = appended_field.as_option().unwrap();
-        let appended_type_name = ts_type_name(appended_field.inner_type(schema), schema);
+        let appended_field = match appended_field.as_option() {
+            Some(option) => option.inner_type(schema),
+            None => match appended_field.as_vec() {
+                Some(vec) => vec.inner_type(schema),
+                None => panic!(
+                    "Appended field must be `Option<T>` or `Vec<T>`: `{}::{}`",
+                    struct_def.name(),
+                    appended_field.name()
+                ),
+            },
+        };
```

Comment:
I think this is a bit clearer:

```suggestion
        let appended_field = match appended_field {
            TypeDef::Option(option_def) => option_def.inner_type(schema),
            TypeDef::Vec(vec_def) => vec_def.inner_type(schema),
            _ => panic!(
                "Appended field must be `Option<T>` or `Vec<T>`: `{}::{}`",
                struct_def.name(),
                appended_field.name()
            ),
        };
```

Reply:
Actually, `appended_field.name()` in the `panic!` should be the field name, not the name of the *type* of the field. So we probably need to call the type `appended_field_type`, so that `appended_field` is still available here as a reference to the `FieldDef`.